lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,38 @@
+module RSpec
+  module Core
+    # @private
+    class NoProfiler
+      def example_groups
+        {}
+      end
+    end
+
+    # @private
+    class Profiler
+      Formatters.register self, :example_group_started, :example_group_finished, :example_started
```

Comment:
Further it can't be merged because the logic is used by 2 formatters, this is removing repeated logic.